### PR TITLE
Remove service named EnableFloatingPointExceptions

### PR DESCRIPTION
**Pull request by wddgit**
This was discussed and decided at the Core meeting
on 30 May 2017. We do not want to use resources maintaining
this service for several reasons. First of all, as far as we
know no one uses or has used this service and it has been
around for more than 10 years. Its unit test is broken. It
does not work properly in the new multithreaded environment
and would be difficult to fix, the precision control
parts of it are obsolete with modern processors, and it
does not work properly with some architectures.

### CODE REVIEW

**Comment by wddgit** [comment]
please test

**Comment by slava77** [comment]
On 6/1/17 7:55 AM, W. David Dagenhart wrote:
> First of all, as far as we
> know no one uses or has used this service and it has been
> around for more than 10 years.

I used it.
The last time was in 2015, and before that in 2009 (at least from the
times that I have a record of).
What is the alternative?

File: Geometry/CMSCommonData/test/runPrint_cfg.py [hunk 1]
```diff
@@ -46,9 +46,6 @@
     firstRun = cms.untracked.uint32(1)
 )
```
Comment (ianna) [inline Geometry/CMSCommonData/test/runPrint_cfg.py]: @wddgit - could you, please, remove the whole file? It's been replaced by SimG4Core/PrintGeomInfo

**Comment by wddgit** [comment]
At the meeting, I was wondering if anyone would say they had used this when the PR came out. I did remove every reference to the service in repository hoping any actual user would notice. As far as I know there is no alternative to the service other than manually coding this functionality yourself for whatever test you want to do. If people think maintaining this service is worth the effort, then it probably would not be very hard to fix the service so enabling floating point exceptions works when the job is configured to have only one thread and only on LINUX. The precision control part is truly obsolete on modern processors and should be deleted. To make it work properly in multithreaded jobs would take some thought. I do not know if it is just difficult or impossible ... Chris may want to comment on this when he returns from vacation on June 12th.

I have not spent time investigating why it fails on AARCH64. It does use glibc functions and has special code enabled for the case #ifdef __APPLE__ so that those functions are defined there ...

**Comment by slava77** [comment]
On 6/1/17 8:32 AM, W. David Dagenhart wrote:
> At the meeting, I was wondering if anyone would say they had used this
> when the PR came out. I did remove every reference to the service in
> repository hoping any actual user would notice. As far as I know there
> is no alternative to the service other than manually coding this
> functionality yourself for whatever test you want to do. If people think
> maintaining this service is worth the effort, then it probably would not
> be very hard to fix the service so enabling floating point exceptions
> works when the job is configured to have only one thread and only on
> LINUX. 

My use cases were related to debugging of issues that eventually
appeared as NaNs
and also some attempts to cleanup cases that trigger FPEs.

**Comment by civanch** [comment]
+1

**Pull request merged**